tests/Unit/FeedXMLGenerationShippingTest.php

```diff
@@ -0,0 +1,530 @@
+<?php
+
+namespace Automattic\WooCommerce\Pinterest\Tests\Unit\Feed;
+
+use ReflectionClass;
+use ShippingHelpers;
+use WC_Product_Variable;
+use \WC_Helper_Product;
+use \WC_Unit_Test_Case;
+
+use Automattic\WooCommerce\Pinterest\Logger;
+use Automattic\WooCommerce\Pinterest\ProductsXmlFeed;
+use Exception;
+
+/**
+ * Feed XML file shipping column generation test class.
+ */
+class Pinterest_Test_Shipping_Feed extends WC_Unit_Test_Case {
+
+	// Holds products for the teardown action.
+	private $products = array();
+
+	/**
+	 * Each test requires at least one product so lets prepare it here.
+	 *
+	 * @return void
+	 */
+	public function setUp() {
+		$this->products[] = WC_Helper_Product::create_simple_product(  true, array( "regular_price" => 15 ) );
+	}
+
+	/**
+	 * Delete products created for testing. And proceed with the cleanup.
+	 *
+	 * @return void
+	 */
+	public function tearDown() {
+		foreach( $this->products as $product ) {
+			$product->delete( true );
+		}
+		ShippingHelpers::cleanup();
+		remove_action( 'woocommerce_flat_rate_shipping_add_rate', array( static::class, 'throwExceptionDuringAddRate' ), 10 );
+	}
+
+	/**
+	 * Check what happens when the user has no shipping zones defined.
+	 *
+	 * @group feed
+	 * @group shipping
+	 */
+	public function testNoShippingZones() {
+		$xml        = $this->ProductsXmlFeed__get_property_g_shipping( end( $this->products ) );
+		$this->assertEquals( '', $xml );
+	}
+
+	/**
+	 * Single shipping zone with free shipping for all destinations.
+	 *
+	 * @group feed
+	 * @group shipping
+	 */
+	public function testZoneWithFreeShipping() {
+		$zone = ShippingHelpers::createZoneWithLocations(
+			[
+				['US', 'country']
+			]
+		);
+		ShippingHelpers::addFreeShipping( $zone );
+
+		$xml = $this->ProductsXmlFeed__get_property_g_shipping( end( $this->products ) );
+		$this->assertEquals( '<g:shipping>US::Free shipping:0.00 USD</g:shipping>', $xml );
+	}
+
+	/**
+	 * Single shipping zone with flat rate shipping.
+	 *
+	 * @group feed
+	 * @group shipping
+	 */
+	public function testZoneWithFlatRateShipping() {
+		$zone = ShippingHelpers::createZoneWithLocations(
+			[
+				['US', 'country']
+			]
+		);
+		ShippingHelpers::addFlatRateShippingMethodToZone( $zone );
+
+		$xml = $this->ProductsXmlFeed__get_property_g_shipping( end( $this->products ) );
+		$this->assertEquals( '<g:shipping>US::Flat rate:15.00 USD</g:shipping>', $xml );
+	}
+
+	/**
+	 * Single shipping zone that has multiple countries and free shipping method.
+	 *
+	 * @group feed
+	 * @group shipping
+	 */
+	public function testZoneWithMultipleCountriesAndFreeShipping(){
+		$zone = ShippingHelpers::createZoneWithLocations(
+			[
+				['US', 'country'],
+				['UK', 'country'],
```

Comment:
The code works correctly - the test is misleading though. I have fixed it in https://github.com/woocommerce/pinterest-for-woocommerce/pull/331/commits/abd1bed82cbe6106561ea275dc85869a3b79e12a The `UK` was not recognized by the code and filtred out. This is why in the results we only had `US` and `IT`. After changing `UK` to `GB` we now had to add it to the results because this is a valid country.

Nice catch.